### PR TITLE
Add a CI GHA workflow

### PR DESCRIPTION
**Pull request by dnadales**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    - name: Store week number as environment variable
+      run: echo "WEEKNUM=$(/usr/bin/date -u '+%W')" >> $GITHUB_ENV
+
+    # Cache the `dist-newstyle` folder.
+    #
+    # See 'restore-cabal-store' for why we use the week number in the primary
+    # key and restore keys.
+    - name: "Cache dist-newstyle"
+      uses: actions/cache@v3
+      if: matrix.ghc!='8.10.7'
```
Comment (amesgen) [inline .github/workflows/ci.yml]:
```suggestion
```

**Review by amesgen — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jorisdral) [reply .github/workflows/ci.yml]: The risk here is that we could get the same flaky build errors that we have in `ouroboros-consensus`. But maybe it's fine to try without the condition for now

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+on:
+  workflow_dispatch:
+  pull_request:
+  merge_group:
+  push:
+    branches:
+      - main
+jobs:
+  build-test-bench-haddocks:
```
Comment (jorisdral) [inline .github/workflows/ci.yml]:
```suggestion
  build:
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jorisdral) [reply .github/workflows/ci.yml]: Never mind, I see that we're only using `ghc-9.2.8`!

**Review by jorisdral — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dnadales) [reply .github/workflows/ci.yml]: I missed that line :facepalm: Thanks!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dnadales) [reply .github/workflows/ci.yml]: Ups. I replaced this with `build-test`. We might add tests soon.

**Pull request merged**